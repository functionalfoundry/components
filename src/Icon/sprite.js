module.exports =  `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="position:absolute; width: 0; height: 0"><defs><path id="size-base-a" d="M4 4h24v24H4z"> </path><mask id="size-base-b" x="0" y="0" width="24" height="24" fill="#fff"> <use xlink:href="#a"></use></mask><path id="size-large-a" d="M2 2h28v28H2z"> </path><mask id="size-large-b" x="0" y="0" width="28" height="28" fill="#fff"> <use xlink:href="#a"></use></mask><path id="size-small-a" d="M6 6h20v20H6z"> </path><mask id="size-small-b" x="0" y="0" width="20" height="20" fill="#fff"> <use xlink:href="#a"></use></mask><path id="size-a" d="M5 26.999h22V5H5z"> </path><mask id="size-b" x="0" y="0" width="22" height="21.999" fill="#fff"> <use xlink:href="#a"></use></mask></defs><symbol viewBox="0 0 32 32" id="add-example"><g fill="none" fill-rule="evenodd"><rect x="7.3" y="7.5" width="18" height="18" rx="3"></rect><path d="M16 12h.5v9H16z"></path><path d="M12 16h9v.5h-9z"></path></g></symbol><symbol viewBox="0 0 22.3105 22.3106" id="add"><path d="M11.155 22.311V0M0 11.155h22.311" data-name="Layer 1" fill="none"></path></symbol><symbol viewBox="64 55 32 32" id="align-center"><path d="M74 62h13m-17 9h20m-14 9h8" fill-rule="evenodd"></path></symbol><symbol viewBox="16 55 32 32" id="align-left"><path d="M22 62h13m-13 9h20m-20 9h8" fill-rule="evenodd"></path></symbol><symbol viewBox="112 55 32 32" id="align-right"><path d="M125 62h13m-20 9h20m-8 9h8" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32 32" id="alignment"><path d="M9 8h14M6 17h20m-14 8h8" fill="none" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 46 46" id="back-hover"><g fill="none"><circle cx="23.001" cy="23" r="22.5" stroke="#059999" stroke-miterlimit="10"></circle><path stroke-miterlimit="10" stroke="#fff" d="M17.301 23.001h11.397"></path><path stroke="#fff" stroke-linejoin="round" d="M22.148 18.153l-4.847 4.848 4.847 4.846"></path></g></symbol><symbol viewBox="0 0 46 46" id="back"><g fill="none"><circle cx="23.001" cy="23" r="22.5" stroke="#0bd8d8" stroke-miterlimit="10"></circle><path stroke-miterlimit="10" stroke="#fff" d="M17.301 23.001h11.397"></path><path stroke="#fff" stroke-linejoin="round" d="M22.148 18.153l-4.847 4.848 4.847 4.846"></path></g></symbol><symbol viewBox="0 0 12.00065 10.5" id="card-like-hover"><path d="M11.81 2.198a3.248 3.248 0 0 0-1.65-1.88A3.097 3.097 0 0 0 8.796 0a3.198 3.198 0 0 0-2.751 1.613.102.102 0 0 0-.023.046l-.057-.024-.001-.002-.005-.01A3.2 3.2 0 0 0 3.203.001a3.097 3.097 0 0 0-1.365.317 3.38 3.38 0 0 0-1.533 4.42c.546 1.203 3.347 3.944 4.968 5.471a1.053 1.053 0 0 0 1.451.002c1.622-1.528 4.423-4.27 4.97-5.473a3.406 3.406 0 0 0 .115-2.54z" fill="#0bd8d8" data-name="Layer 1"></path></symbol><symbol viewBox="0 0 12.00068 10.5" id="card-like"><path class="cls-1" d="M8.796 1a2.097 2.097 0 0 1 .925.217 2.254 2.254 0 0 1 1.144 1.31 2.41 2.41 0 0 1-.083 1.798c-.265.583-1.65 2.244-4.743 5.158l.686.728-.766-.73c-3.095-2.916-4.48-4.575-4.743-5.156a2.373 2.373 0 0 1 1.06-3.107A2.1 2.1 0 0 1 3.202 1 2.198 2.198 0 0 1 5.08 2.103a1.004 1.004 0 0 0 .083.13.993.993 0 0 0 .415.324l.057.024a1 1 0 0 0 1.288-.49A2.196 2.196 0 0 1 8.796 1m0-1a3.198 3.198 0 0 0-2.751 1.613.103.103 0 0 0-.023.046l-.057-.024-.002-.002-.004-.01A3.2 3.2 0 0 0 3.203.002a3.097 3.097 0 0 0-1.365.317 3.38 3.38 0 0 0-1.533 4.42c.546 1.203 3.347 3.944 4.968 5.471a1.053 1.053 0 0 0 1.452.002c1.621-1.527 4.422-4.27 4.968-5.472a3.402 3.402 0 0 0 .116-2.54 3.246 3.246 0 0 0-1.65-1.88A3.093 3.093 0 0 0 8.797 0z" fill="#879399" data-name="Layer 1"></path></symbol><symbol viewBox="0 0 16 16" id="caret"><path d="M14.17 4.1l-6.099 8.042-6.1-8.043" fill="none"></path></symbol><symbol viewBox="0 0 19.25889 19.25793" id="close-hover"><path d="M18.905.354L.354 18.904m18.551 0L.354.354" data-name="Layer 1" fill="none" stroke="#0bd8d8" stroke-miterlimit="10"></path></symbol><symbol viewBox="0 0 19.25889 19.25793" id="close"><path d="M18.905.354L.354 18.904m18.551 0L.354.354"></path></symbol><symbol viewBox="256 622 16 16" id="column-sort-asc"><g fill="none" opacity=".64" transform="matrix(1 0 0 -1 256 638)"><circle fill-opacity=".16" fill="#FFF" fill-rule="evenodd" cx="8.5" cy="7.5" r="7.5"></circle><path stroke="#FFF" d="M5 7l3.5 4L12 7"></path><path d="M8.5 10.5v-7" stroke="#FFF" stroke-linecap="square"></path></g></symbol><symbol viewBox="232 624 16 16" id="column-sort-desc"><g fill="none" opacity=".64" transform="translate(232 624)"><circle fill-opacity=".16" fill="#FFF" fill-rule="evenodd" cx="8.5" cy="7.5" r="7.5"></circle><path stroke="#FFF" d="M5 7l3.5 4L12 7"></path><path d="M8.5 10.5v-7" stroke="#FFF" stroke-linecap="square"></path></g></symbol><symbol viewBox="0 0 32 32" id="delete"><path d="M21.701 13v10.703c0 .55-.45 1-1 1H11c-.55 0-1-.45-1-1V13m3 9.939V14m3 8.939V14m3 8.939V14M8 11.017h15.619V9H8zm4-2.14h7.307V7H12z" fill="none" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32 32" id="duplicate"><g fill="none" fill-rule="evenodd"><path d="M9 16.187H7c-.55 0-1-.45-1-1V7c0-.549.45-1 1-1h8.186c.549 0 1 .451 1 1v2"></path><path d="M25.109 24.112c0 .55-.45 1-1 1H12c-.55 0-1-.45-1-1V12c0-.549.45-1 1-1h12.109c.55 0 1 .451 1 1v12.112zM18 14v8.03M22.029 18H14"></path></g></symbol><symbol viewBox="0 0 32 32" id="github"><path d="M15.998 3C8.821 3 3 8.82 3 16c0 5.743 3.723 10.616 8.89 12.335.65.118.888-.283.888-.626 0-.309-.012-1.126-.018-2.212-3.615.786-4.378-1.741-4.378-1.741-.591-1.503-1.444-1.903-1.444-1.903-1.181-.807.09-.79.09-.79 1.304.092 1.991 1.34 1.991 1.34 1.159 1.986 3.043 1.413 3.783 1.08.118-.84.453-1.413.824-1.738-2.886-.33-5.921-1.443-5.921-6.425 0-1.419.507-2.58 1.338-3.49-.134-.327-.578-1.649.129-3.439 0 0 1.09-.35 3.574 1.333A12.474 12.474 0 0 1 16 9.286a12.47 12.47 0 0 1 3.254.438c2.482-1.682 3.57-1.333 3.57-1.333.711 1.79.265 3.112.13 3.44.834.91 1.337 2.07 1.337 3.489 0 4.993-3.04 6.093-5.935 6.415.467.402.883 1.194.883 2.408 0 1.737-.016 3.14-.016 3.566 0 .347.234.752.894.625C25.278 26.611 29 21.742 29 16.001 29 8.82 23.178 3 15.998 3" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32 32" id="grid"><path d="M6.5 6.5h19v7h-19zm0 12h7v7h-7zm12 0h7v7h-7z" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 18.4628 15.1128" id="layout-hover"><g data-name="Layer 1" fill="none" stroke="#0bd8d8" stroke-miterlimit="10"><path d="M17.963 12.298v1.315a1.003 1.003 0 0 1-1 1H1.5a1.003 1.003 0 0 1-1-1v-1.315"></path><rect x=".5" y=".5" width="17.463" height="7.056" rx="1" ry="1"></rect></g></symbol><symbol viewBox="1080 144 32 32" id="layout"><path d="M1106.015 152.419a.628.628 0 0 0-.625-.631h-19.34a.63.63 0 0 0-.625.63v6.38c0 .347.283.63.626.63h19.34a.628.628 0 0 0 .624-.63v-6.38zm1.251 0v6.378a1.887 1.887 0 0 1-1.876 1.893h-19.34a1.887 1.887 0 0 1-1.875-1.893v-6.378c0-1.045.841-1.893 1.876-1.893h19.34c1.033 0 1.875.848 1.875 1.893zm-1.25 13.621h1.25v1.66a1.887 1.887 0 0 1-1.876 1.893h-19.34a1.887 1.887 0 0 1-1.875-1.893v-1.66h1.25v1.66a.63.63 0 0 0 .626.631h19.34a.628.628 0 0 0 .624-.63v-1.661z" fill="#FFF" opacity=".64" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32.82208 31.50086" id="logo"><g data-name="Layer 1"><path d="M8.955 31.5a1.5 1.5 0 0 1-1.45-1.12L.049 1.88a1.5 1.5 0 1 1 2.903-.76l7.455 28.5a1.501 1.501 0 0 1-1.452 1.88zm15.291-.048a1.5 1.5 0 0 0 1.072-1.831l-7.456-28.5a1.5 1.5 0 1 0-2.902.76l7.455 28.5a1.498 1.498 0 0 0 1.831 1.07z" fill="#0bd8d8"></path><path d="M16.411 7.428L10.506 30l-1.55-5.927L14.86 1.5zM31.701.049a1.5 1.5 0 0 0-1.83 1.072l-6.005 22.952 1.55 5.928L32.774 1.88A1.5 1.5 0 0 0 31.7.05z" fill="#00b5b5"></path><path d="M24.9 20.122l1.55 5.927L25.418 30l-1.55-5.927zM8.956 24.073l1.55 5.928 1.034-3.952-1.55-5.927z" opacity=".1"></path></g></symbol><symbol viewBox="174 718 20 20" id="minus"><path d="M191.778 728h-15.556" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32 32" id="more-horizontal"><path d="M7.915 14c1.058 0 1.915.895 1.915 2 0 1.104-.857 2-1.915 2S6 17.104 6 16c0-1.105.857-2 1.915-2M16 14c1.058 0 1.915.895 1.915 2 0 1.104-.857 2-1.915 2s-1.915-.896-1.915-2c0-1.105.857-2 1.915-2m8.085 0c1.057 0 1.915.895 1.915 2 0 1.104-.858 2-1.915 2-1.059 0-1.915-.896-1.915-2 0-1.105.856-2 1.915-2" fill="none"></path></symbol><symbol viewBox="0 0 50 50" id="open"><g data-name="Layer 1"><circle cx="25" cy="25" r="25" fill="#293840"></circle><path fill="none" stroke="#0bd8d8" stroke-miterlimit="10" d="M16.301 25h17.398"></path><path fill="none" stroke="#0bd8d8" stroke-linejoin="round" d="M25 16.301L33.699 25 25 33.699"></path></g></symbol><symbol viewBox="168 760 32 32" id="primary-minus-hover"><circle fill="#0F1619" fill-rule="evenodd" cx="184" cy="776" r="16"></circle><path fill="#07b3b3" d="M177 775.5h14v1h-14z" fill-rule="evenodd"></path></symbol><symbol viewBox="168 760 32 32" id="primary-minus"><circle fill="#0F1619" fill-rule="evenodd" cx="184" cy="776" r="16"></circle><path fill="#0BE5E5" d="M177 775.5h14v1h-14z" fill-rule="evenodd"></path></symbol><symbol viewBox="168 760 32 32" id="primary-plus-hover"><circle fill="#0F1619" fill-rule="evenodd" cx="184" cy="776" r="16"></circle><path d="M184.5 775.5V769h-1v6.5H177v1h6.5v6.5h1v-6.5h6.5v-1h-6.5z" fill="#07b3b3" fill-rule="evenodd"></path></symbol><symbol viewBox="168 760 32 32" id="primary-plus"><circle fill="#0F1619" fill-rule="evenodd" cx="184" cy="776" r="16"></circle><path d="M184.5 775.5V769h-1v6.5H177v1h6.5v6.5h1v-6.5h6.5v-1h-6.5z" fill="#0BE5E5" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 20.85347 20.85363" id="search"><g data-name="Layer 1" fill="none" stroke="#fff" stroke-miterlimit="10"><circle cx="11.078" cy="9.777" r="9.276"></circle><path d="M4.519 16.336L.353 20.5"></path></g></symbol><symbol viewBox="0 0 21 14" id="share"><path d="M20.72 6.692L11.789 0v3.443C6.404 3.971 1.406 7.658 0 13.872c1.821-3.991 7.067-4.697 11.341-4.697.08 0 .157.012.236.014.071-.002.141-.012.212-.013v3.546l8.931-6.03z" fill="none"></path></symbol><symbol viewBox="659 74 32 32" id="size-base"><g transform="translate(659 74)"><use mask="url(#size-base-b)" xlink:href="#size-base-a"></use><path d="M25 7.5V7h-7v1h6v6h1V7.5zM8 24h6v1H7v-7h1v6z" fill-rule="evenodd"></path><path fill-rule="evenodd" d="M24.036 7.258l.707.707-5.889 5.888-.707-.707zm-11 11l.707.707-5.889 5.888-.707-.707z"></path></g></symbol><symbol viewBox="663 55 32 32" id="size-large"><g transform="translate(663 55)"><use mask="url(#size-large-b)" xlink:href="#size-large-a"></use><path d="M7 25h6v1H6v-7h1v6z" fill-rule="evenodd"></path><path fill-rule="evenodd" d="M11.621 19.672l.707.707-4.95 4.95-.706-.708zM7 7h6V6H6v7h1V7z"></path><path fill-rule="evenodd" d="M11.621 12.328l.707-.707-4.95-4.95-.706.708zM25 25h-6v1h7v-7h-1v6z"></path><path fill-rule="evenodd" d="M20.379 19.672l-.707.707 4.95 4.95.706-.708zM25 7h-6V6h7v7h-1V7z"></path><path fill-rule="evenodd" d="M20.379 12.328l-.707-.707 4.95-4.95.706.708z"></path></g></symbol><symbol viewBox="606 46 32 32" id="size-small"><g fill="none" transform="translate(606 46)"><use mask="url(#size-small-b)" xlink:href="#size-small-a"></use><path d="M21 11.5V11h-7v1h6v6h1v-6.5z"></path><path d="M20.036 11.258l.707.707-5.889 5.888-.707-.707z"></path></g></symbol><symbol viewBox="0 0 40 40" id="size"><g transform="translate(4 4)" fill="none" fill-rule="evenodd"><path d="M8.5 16.5v7h7M9 23.034l5.04-5.04M23.5 15.5v-7h-7m6.5.466l-5.04 5.04"></path><use mask="url(#size-b)" xlink:href="#size-a"></use></g></symbol><symbol viewBox="0 0 14 14" id="star"><path d="M7 0l1.653 5.348H14L9.674 8.653 11.326 14l-4.325-3.304L2.674 14l1.652-5.347L0 5.348h5.348z" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 32 32" id="theme"><path d="M10.356 16.607c0-.37.141-.74.422-1.02l5.562-5.566a1.443 1.443 0 0 1 2.043 0l5.563 5.565c.283.281.424.65.424 1.021H10.356zm14.662-2.122L19.454 8.92a2.958 2.958 0 0 0-1.286-.756l.871-.87a.759.759 0 1 0-1.073-1.07l-3.61 3.611a.238.238 0 0 0-.041.043l-4.607 4.607a2.953 2.953 0 0 0 0 4.182l5.564 5.566a2.958 2.958 0 0 0 4.182 0l5.564-5.566a2.954 2.954 0 0 0 0-4.182zM6 24.466c0 1.391 1.065 2.528 2.367 2.528s2.367-1.137 2.367-2.528c0-1.388-2.367-5.047-2.367-5.047S6 23.08 6 24.466z" fill="none"></path></symbol><symbol viewBox="0 0 24 24" id="twitter"><path d="M23.444 4.834c-.814.363-1.5.375-2.228.016.938-.562.981-.957 1.32-2.019-.878.521-1.851.9-2.886 1.104a4.545 4.545 0 0 0-7.742 4.145 12.897 12.897 0 0 1-9.366-4.748 4.525 4.525 0 0 0-.615 2.285c0 1.577.803 2.967 2.021 3.782a4.527 4.527 0 0 1-2.057-.568l-.001.057a4.547 4.547 0 0 0 3.646 4.456 4.575 4.575 0 0 1-2.053.079 4.551 4.551 0 0 0 4.245 3.155A9.138 9.138 0 0 1 1 18.459 12.865 12.865 0 0 0 7.966 20.5c8.358 0 12.928-6.924 12.928-12.929 0-.198-.003-.393-.012-.588.887-.64 1.953-1.237 2.562-2.149z"></path></symbol><symbol viewBox="760 144 32 32" id="undo"><path d="M779.404 170.523c-5.143 0-9.328-4.183-9.328-9.325h1.023c0 4.579 3.726 8.302 8.305 8.302 4.578 0 8.302-3.723 8.302-8.302 0-4.58-3.724-8.304-8.302-8.304h-14.446l2.147 2.147-.724.722-3.381-3.38 3.38-3.383.725.724-2.147 2.148h14.446c5.143 0 9.324 4.182 9.324 9.326 0 5.142-4.181 9.325-9.324 9.325z" fill="#FFF" fill-rule="evenodd"></path></symbol><symbol viewBox="0 0 26.2607 27.07023" id="widget-feed"><g data-name="Layer 1" fill="none" stroke-miterlimit="10"><path stroke="#0bd8d8" d="M0 14.706h3.223L8.121 1.847l4.265 23.178 4.624-14.393h9.251"></path><path stroke="#fff" d="M26.261 14.913h-8.038m8.038-8.587h-8.038m8.038 12.71h-8.038"></path></g></symbol></svg>`; // eslint-disable-line