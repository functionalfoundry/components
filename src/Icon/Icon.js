/* @flow */
import React from 'react'
import Theme from 'js-theme'
import View from '../View'
import sprite from './sprite'

type SizeT = 'tiny' | 'small' | 'medium' | 'large' | 'huge'

type PropsT = {
  children: React.Children,
  fill: string,
  name: string,
  size: SizeT,
  stroke: string,
  theme: Object,
}

class Icon extends React.Component {
  static defaultProps = {
    size: 'small',
    theme: defaultTheme,
  }

  componentWillMount() {
    if (document && !document.getElementById('sprite')) {
      inject(sprite)
    }
  }

  props: PropsT

  render() {
    const {
      name,
      children,
      fill,
      stroke,
      size,
      theme,
      ...props,
    } = this.props

    return (
      <View
        {...theme.icon}
        {...props}
        inline
      >
        {
          isInline(children)
          ? <svg {...theme.svg}>{children}</svg>
          : <svg {...theme.svg}><use xlinkHref={`#${name}`} /></svg>
        }
      </View>
    )
  }
}

const isInline = (children: React.Children) =>
  children && (children.type === 'svg')

const inject = (content: string) => {
  let svg = document.getElementById('sprite')
  // TODO: Right now this is injecting on every load!
  // Figure out how to test preloading the SVG for code coverage
  // if (!svg) {
  svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
  svg.setAttribute('width', '0')
  svg.setAttribute('height', '0')
  svg.setAttribute('style', 'display:none;')
  svg.setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:xlink', 'http://www.w3.org/1999/xlink')
  svg.id = 'sprite'
  document.body.appendChild(svg)

  const wrapper = document.createElement('div')
  wrapper.innerHTML = `<svg>${content}</svg>`
  Array.prototype.slice
    .call(wrapper.childNodes[0].childNodes)
    .forEach((el: HTMLElement) => document.getElementById('sprite').appendChild(el))
  // }
}

// const capitalize = (str: string) => str.charAt(1).toUpperCase() + str.slice(1)

const defaultTheme = ({
  fill,
  stroke,
  size,
}: PropsT) => ({
  icon: {
    ...getFillStyle(fill),
    ...getStrokeStyle(stroke),
    lineHeight: 0,
  },
  svg: {
    ...getSizeStyle(size),
  },
})

const getFillStyle = (fill: string) => (fill ? { fill } : {})
const getStrokeStyle = (stroke: string) => (stroke ? { stroke } : {})

const getSizeStyle = (size: SizeT) => {
  switch (size) {
    case 'tiny':
      return {
        width: 12,
        height: 12,
      }
    case 'small':
      return {
        width: 16,
        height: 16,
      }
    case 'large':
      return {
        width: 36,
        height: 36,
      }
    case 'huge':
      return {
        width: 48,
        height: 48,
      }
    case 'medium':
    default:
      return {
        width: 24,
        height: 24,
      }
  }
}

export default Theme('Icon', defaultTheme)(Icon)
