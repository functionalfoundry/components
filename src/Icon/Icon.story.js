import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Icon from '.'

storiesOf('Icon', module)
  .add('Regular', () => (
    <div
      style={styles.container}
    >
      <Icon
        name='logo'
        size='huge'
      />
    </div>
  ))

const styles = {
  container: {
    width: '100%',
  },
}
