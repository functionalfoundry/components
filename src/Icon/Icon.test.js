import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '.'

test('Default', () => {
  const icon = renderer.create(
    <Icon />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Tiny', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='tiny'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Small', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='small'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Medium', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='medium'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Large', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='large'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Huge', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='huge'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('Unknown size', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      size='unknown'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('With child SVG', () => {
  const icon = renderer.create(
    <Icon>
      <svg>
        <g />
      </svg>
    </Icon>
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('With fill color', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      fill='#cdcdcd'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('With stroke color', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      stroke='#cdcdcd'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})

test('With sprite already injected', () => {
  const icon = renderer.create(
    <Icon
      name='logo'
      stroke='#cdcdcd'
    />
  ).toJSON()
  expect(icon).toMatchSnapshot()
})
