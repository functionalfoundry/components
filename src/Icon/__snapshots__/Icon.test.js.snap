exports[`test Default 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}>
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#undefined" />
  </svg>
</span>
`;

exports[`test Huge 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="huge">
  <svg
    className="svg_w39vrk">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test Large 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="large">
  <svg
    className="svg_r2sz10">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test Medium 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="medium">
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test Small 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="small">
  <svg
    className="svg_1r8j7bt">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test Tiny 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="tiny">
  <svg
    className="svg_j1577k">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test Unknown size 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}
  size="unknown">
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test With child SVG 1`] = `
<span
  className="view_18uv8rq icon_120drhm"
  data-test-id={undefined}>
  <svg
    className="svg_bqvwbh">
    <svg>
      <g />
    </svg>
  </svg>
</span>
`;

exports[`test With fill color 1`] = `
<span
  className="view_18uv8rq icon_1w7h16b"
  data-test-id={undefined}
  fill="#cdcdcd">
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test With sprite already injected 1`] = `
<span
  className="view_18uv8rq icon_16r926a"
  data-test-id={undefined}
  stroke="#cdcdcd">
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;

exports[`test With stroke color 1`] = `
<span
  className="view_18uv8rq icon_16r926a"
  data-test-id={undefined}
  stroke="#cdcdcd">
  <svg
    className="svg_bqvwbh">
    <use
      xlinkHref="#logo" />
  </svg>
</span>
`;
